src/microbit/model/image.py

```diff
@@ -87,19 +156,27 @@ def copy(self):
     # ie: Pixel that is at brightness 4 would become brightness 5
     # and pixel that is at brightness 9 would become brightness 0.
     def invert(self):
+        if self.read_only:
+            raise TypeError(CONSTANTS.COPY_ERR_MESSAGE)
+

```

Comment:
If I'm not missing something, you could probably remove the check in most methods since you already check this in `set_pixel`?

Reply:
very true! thanks :D